sqlmesh/core/signal.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+import abc
+import typing as t
+from sqlmesh.core._typing import Batch
```

Comment:
Is this something users suppose to import? In this case let's probably export this in the `sqlmesh/__init__.py`. Also let's put this under `if t.TYPE_CHECKING`

Reply:
I believe users would need `Signal` and `Batch` (for its method signature). These are all concerns of the `scheduler` (where `Batch` used to be defined). I've created a `signal.py` file, but that seems unnecessary and it's causing some `import` pains.

Any objection to moving `Signal` and `SignalFactory` from `signal.py` into `scheduler.py`? Then we can re-export whatever is necessary from `sqlmesh/__init__.py` to simplify users' life.

Reply:
Took the liberty to move things around in https://github.com/TobikoData/sqlmesh/pull/2822/commits/f1aa0455f7442d974076f9380e5906814e48a2b2 PTAL